korman/korlib/__init__.py

```diff
@@ -101,3 +103,35 @@ def inspect_wavefile(stream, header):
         header.read(stream)
 
         return chunks[b"data"]["size"]
+
+    def is_legal_python2_identifier(identifier):
```

Comment:
I think using the [built-in function for this](https://docs.python.org/3/library/stdtypes.html#str.isidentifier) might be sufficient.  According to [the docs](https://docs.python.org/3/reference/lexical_analysis.html#identifiers), the only additional restriction in Python 2 is that the string must also be [within the ASCII range](https://docs.python.org/3/library/stdtypes.html#str.isascii):

> Within the ASCII range (U+0001..U+007F), the valid characters for identifiers are the same as in Python 2.x: the uppercase and lowercase letters A through Z, the underscore _ and, except for the first character, the digits 0 through 9.
> 
> Python 3.0 introduces additional characters from outside the ASCII range (see PEP 3131). For these characters, the classification uses the version of the Unicode Character Database as included in the unicodedata module.

In which case, I believe this entire function can be reduced to:

`return str.isidentifier(identifier) and str.isascii(identifier)`

Reply:
Unfortunately, we are presently constrained to Python 3.5. `identifier.isascii()` was added in Python 3.7. But, also, in the documentation right next door is a note about keywords. I did forget to ensure someone isn't trying to name their age `class`, `def`, `if`, `else`, `yield`, etc...

Reply:
Ah, strange that `isascii` is so new.  Still, that leaves a lot of simplification!

Reply:
Isn’t that just a more convoluted way of saying
```python
if any(ord(i) > 0x7F for i in identifier):
    return False
```
?